src/Extension/BetterNavigatorExtension.php

```diff
@@ -111,10 +111,19 @@ private function generateNavigator()
                 }
             }
         }
-        // Only show edit link if user has permission to edit this page
-        $editLink = array_key_exists('CMSLink', $nav)
-            && ($isDev || $this->owner->dataRecord->canEdit() && Permission::check('CMS_ACCESS_CMSMain'))
-            ? $nav['CMSLink']['Link'] : false;
+        
+        // Only show edit link if user has CMS access
+        if($isDev || Permission::check('CMS_ACCESS_CMSMain')) {
+            // Check for edit link override, e.g. for a DataObject
+            if(method_exists($this->owner, 'BetterNavigatorEditLink')) {
+                $editLink = $this->owner->BetterNavigatorEditLink();
+            } else {
+                // Only show edit link if user has permission to edit this page
+                $editLink = array_key_exists('CMSLink', $nav)
+                && ($this->owner->dataRecord->canEdit())
```

Comment:
This PR is causing a regression - if you're in dev mode but not logged in the 'edit in CMS' button is no longer visible. Need a tweak here:
```suggestion
                && ($isDev || $this->owner->dataRecord->canEdit())
```